test_dask_da_groupby_quantile not passing

### What happened?

The test fails because the expected `ValueError` is not raised:
```
________________________ test_dask_da_groupby_quantile _________________________
[gw2] linux -- Python 3.13.0 /usr/bin/python3

    @requires_dask
    def test_dask_da_groupby_quantile() -> None:
        # Only works when the grouped reduction can run blockwise
        # Scalar quantile
        expected = xr.DataArray(
            data=[2, 5], coords={"x": [1, 2], "quantile": 0.5}, dims="x"
        )
        array = xr.DataArray(
            data=[1, 2, 3, 4, 5, 6], coords={"x": [1, 1, 1, 2, 2, 2]}, dims="x"
        )
>       with pytest.raises(ValueError):
E       Failed: DID NOT RAISE <class 'ValueError'>

/builddir/build/BUILD/python-xarray-2024.11.0-build/BUILDROOT/usr/lib/python3.13/site-packages/xarray/tests/test_groupby.py:295: Failed
```

### What did you expect to happen?

Test pass.

### Minimal Complete Verifiable Example

```Python
pytest 'tests/test_groupby.py::test_dask_da_groupby_quantile'
```


### MVCE confirmation

- [X] Minimal example — the example is as focused as reasonably possible to demonstrate the underlying issue in xarray.
- [X] Complete example — the example is self-contained, including all data and the text of any traceback.
- [X] Verifiable example — the example copy & pastes into an IPython prompt or [Binder notebook](https://mybinder.org/v2/gh/pydata/xarray/main?urlpath=lab/tree/doc/examples/blank_template.ipynb), returning the result.
- [X] New issue — a search of GitHub Issues suggests this is not a duplicate.
- [X] Recent environment — the issue occurs with the latest version of xarray and its dependencies.

### Relevant log output

_No response_

### Anything else we need to know?

_No response_

### Environment

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.13.0 (main, Oct  8 2024, 00:00:00) [GCC 14.2.1 20240912 (Red Hat 14.2.1-4)]
python-bits: 64
OS: Linux
OS-release: 6.11.5-200.fc40.x86_64
machine: x86_64
processor: 
byteorder: little
LC_ALL: None
LANG: C.UTF-8
LOCALE: ('C', 'UTF-8')
libhdf5: 1.14.5
libnetcdf: 4.9.2

xarray: 2024.11.0
pandas: 2.2.1
numpy: 1.26.4
scipy: 1.14.1
netCDF4: 1.7.2
pydap: None
h5netcdf: None
h5py: None
zarr: 2.18.3
cftime: 1.6.4
nc_time_axis: None
iris: None
bottleneck: 1.3.7
dask: 2024.11.2
distributed: None
matplotlib: 3.9.1
cartopy: None
seaborn: 0.13.2
numbagg: None
fsspec: 2024.10.0
cupy: None
pint: 0.24.4
sparse: None
flox: None
numpy_groupies: None
setuptools: 74.1.3
pip: 24.3.1
conda: None
pytest: 8.3.3
mypy: None
IPython: None
sphinx: 7.3.7
</details>

--- a/xarray/tests/test_groupby.py
+++ b/xarray/tests/test_groupby.py
@@ -284,7 +284,6 @@ def test_da_groupby_empty() -> None:
 
 @requires_dask
 def test_dask_da_groupby_quantile() -> None:
-    # Only works when the grouped reduction can run blockwise
     # Scalar quantile
     expected = xr.DataArray(
         data=[2, 5], coords={"x": [1, 2], "quantile": 0.5}, dims="x"
@@ -292,8 +291,6 @@ def test_dask_da_groupby_quantile() -> None:
     array = xr.DataArray(
         data=[1, 2, 3, 4, 5, 6], coords={"x": [1, 1, 1, 2, 2, 2]}, dims="x"
     )
-    with pytest.raises(ValueError):
-        array.chunk(x=1).groupby("x").quantile(0.5)
 
     # will work blockwise with flox
     actual = array.chunk(x=3).groupby("x").quantile(0.5)
@@ -327,7 +324,8 @@ def test_dask_da_groupby_median() -> None:
     assert_identical(expected, actual)
 
 
-def test_da_groupby_quantile() -> None:
+@pytest.mark.parametrize("use_flox", [True, False])
+def test_da_groupby_quantile(use_flox) -> None:
     array = xr.DataArray(
         data=[1, 2, 3, 4, 5, 6], coords={"x": [1, 1, 1, 2, 2, 2]}, dims="x"
     )
@@ -336,8 +334,10 @@ def test_da_groupby_quantile() -> None:
     expected = xr.DataArray(
         data=[2, 5], coords={"x": [1, 2], "quantile": 0.5}, dims="x"
     )
-    actual = array.groupby("x").quantile(0.5)
-    assert_identical(expected, actual)
+
+    with xr.set_options(use_flox=use_flox):
+        actual = array.groupby("x").quantile(0.5)
+        assert_identical(expected, actual)
 
     # Vector quantile
     expected = xr.DataArray(
@@ -345,7 +345,8 @@ def test_da_groupby_quantile() -> None:
         coords={"x": [1, 2], "quantile": [0, 1]},
         dims=("x", "quantile"),
     )
-    actual = array.groupby("x").quantile([0, 1])
+    with xr.set_options(use_flox=use_flox):
+        actual = array.groupby("x").quantile([0, 1])
     assert_identical(expected, actual)
 
     array = xr.DataArray(
@@ -356,7 +357,8 @@ def test_da_groupby_quantile() -> None:
         e = [np.nan, 5] if skipna is False else [2.5, 5]
 
         expected = xr.DataArray(data=e, coords={"x": [1, 2], "quantile": 0.5}, dims="x")
-        actual = array.groupby("x").quantile(0.5, skipna=skipna)
+        with xr.set_options(use_flox=use_flox):
+            actual = array.groupby("x").quantile(0.5, skipna=skipna)
         assert_identical(expected, actual)
 
     # Multiple dimensions

--- a/xarray/tests/test_groupby.py
+++ b/xarray/tests/test_groupby.py
@@ -324,8 +324,8 @@ def test_dask_da_groupby_median() -> None:
     assert_identical(expected, actual)
 
 
-@pytest.mark.parametrize("use_flox", [True, False])
-def test_da_groupby_quantile(use_flox) -> None:
+@pytest.mark.parametrize("use_flox", [pytest.param(True, marks=requires_flox), False])
+def test_da_groupby_quantile(use_flox: bool) -> None:
     array = xr.DataArray(
         data=[1, 2, 3, 4, 5, 6], coords={"x": [1, 1, 1, 2, 2, 2]}, dims="x"
     )

